### PR TITLE
RELEASE 1.2 ✨

### PR DESCRIPTION
**Pull request by lwasser**
# Stravalib Release Pull Request Template

Please use this template when you are preparing to make a release to stravalib

* [An overview of our release workflow can be found in our documentation.](https://stravalib.readthedocs.io/contributing/build-release-guide.md)
* [Before making a release be sure to check out test PyPI](https://pypi.org/project/stravalib/) to ensure that the build is working properly.

## Release checklist
- [x] Be sure to clearly specify what version you are bumping to in the PR title: Example: Bump to version x.x
- [x] Add the version of this release to our changelog
- [x] Organize items changes under the new version in groups as follows: Added, Fixed and Changed
- [x] Add all contributors to the release below those sections
- [x] Wait for a maintainer to approve the pull request. Then merge! You are now ready to create the release.

Once this PR is merged you are ready to

- [ ] Create a tagged release on GitHub using the same version that you merged in the changelog added here
- [ ] When you publish that release, the GitHub action to push to PyPI will be invoked.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,13 +1,22 @@
 # Change Log
 
 ## Unreleased
-* Fix: Internal warnings should be ignored in tests (@jsamoocha, #319)
+
+## v1.2.0
+
+### Added
```
Comment (lwasser) [inline changelog.md:7]: oh wait - i'm confusing myself. i shouldn't work when i'm tired. the other changelog in the pydantic branch has those commits. this looks ok to me. but i'd love a second look as i'm obviously not processing information correctly today 😆 i just want to ensure all of these items are associated with what is on master which they should be.

Comment (jsamoocha) [reply changelog.md:7]: Whenever we merged a PR on `master`, I rebased and force-pushed `pydantic_model`. This ensures the changelog in the latter branch is complete whenever we merge that into `master`.

I don't think there's anything to worry about re. the changelog (but I understand the confusion :-), I guess that's part of long-lasting feature branches (which I'm not a fan of, but was kinda inevitable in this case))

**Comment by jsamoocha** [comment]
Can you please rebase on `master` to include #336 in this release? I can then rebase `pydantic_model` on 1.2.0 and ask the community to test it.

File: changelog.md (hunk 1 above)
Comment (lwasser) [reply changelog.md:7]: ok perfect! i just had a moment but then i recognized i was tired last night.!!

**Comment by lwasser** [comment]
ok @jsamoocha just rebased and updated here. lemme know if you catch anything else that looks off!

**Review by jsamoocha — APPROVED** [review]
Great, let's release 1.2.0!

**Comment by lwasser** [comment]
yay! ok merging this @jsamoocha ... i'll test the push to test pypi tonight and then create a release then.

**Pull request merged**